File: QuizGame.ManuelE-Osorio/QuizGame-Client/src/app/models/quiz.ts
import { FormControl } from "@angular/forms";
import { Question } from "./question";
import { User } from "./user";

export interface Quiz{
    id: number;
    name: string;
    description: string;
    questions: Question[];
    gamesId: number[];
    gamesName: number[];
    owner: User;
}

export interface QuizForm{
    id?: FormControl<number|null>;
    name: FormControl<string|null>;
    description: FormControl<string|null>;
}

File: QuizGame.ManuelE-Osorio/QuizGame-WebAPI/Models/Quiz.cs
using System.ComponentModel.DataAnnotations;

namespace QuizGame.Models;

public class Quiz
{
    public int Id {get; set;}
    
    [Required, StringLength(100, MinimumLength = 3)]
    public string? Name {get; set;}
    public string? Description {get; set;}
    public ICollection<Question>? Questions {get; set;}
    public ICollection<Game>? Games {get; set;}
    public QuizGameUser? Owner {get; set;}

    private Quiz()
    {
        Name = default!;
        Questions = default!;
    }
    public Quiz(string name, ICollection<Question> questions) 
    {
        Name = name;
        Questions = questions;
    }

    public Quiz(QuizDto quizDto)
    {
        Id = quizDto.Id;
        Name = quizDto.Name;
        Description = quizDto.Description;
    }
}